oonib/testhelpers/http_helpers.py

```diff
@@ -390,6 +394,9 @@ def tcp_connect(self, socket):
         except ConnectionRefusedError:
             socket_info['status'] = False
             socket_info['failure'] = 'connection_refused_error'
+        except ConnectError:
+            socket_info['status'] = False
+            socket_info['failure'] = 'connect_error'
```

Comment:
Yes I agree that we should do a better job at extracting the error message, however mapping them to something that is actually parseable is a bit of work and it would need to be done in both the probe and in the backend.

The actual error string is set, I believe, by the underlying `socket` and not by twisted itself so that is where we would have to go hunting for the error strings.